src/transformers/models/bart/modeling_tf_bart.py

```diff
@@ -1373,29 +1402,99 @@ def prepare_inputs_for_generation(
         decoder_input_ids,
         past=None,
         attention_mask=None,
+        decoder_attention_mask=None,
         head_mask=None,
         decoder_head_mask=None,
         cross_attn_head_mask=None,
         use_cache=None,
         encoder_outputs=None,
         **kwargs
     ):
+
         # cut decoder_input_ids if past is used
         if past is not None:
             decoder_input_ids = decoder_input_ids[:, -1:]
 
+        if decoder_attention_mask is not None:  # xla
+            decoder_position_ids = tf.math.cumsum(decoder_attention_mask, axis=-1, exclusive=True)[:, -1:]
```

Comment:
I will try to understand `_update_model_kwargs_for_xla_generation` later. 

However, I am wondering if `decoder_attention_mask` will be always `None` if the mode is **NOT** in `XLA`, at least, with our current usage/implementation of `generate`.

The reason I am asking is: the computation of `(decoder_position_ids) using `tf.math.cumsum` actually will give different results compared to the current version.

For example, considering `decoder_attention_mask` to be `[1, 1, 1, 0, 0]`.
So far, we don't have `position_ids`, and `TFBartLearnedPositionalEmbedding` will compute `[0, 1, 2, 3, 4]` as positions.

With this PR, it would be `[0, 1, 2, 2, 2]` I think.

Despite we have attention mechanism, I am always feeling uncomfortable to see the inputs being different.